docs/articles/documentation/reference/test-api/testcontroller/dispatchEvent.md

```diff
@@ -0,0 +1,152 @@
+---
+layout: docs
+title: t.dispatchEvent Method
+permalink: /documentation/reference/test-api/testcontroller/dispatchevent.html
+---
+# t.dispatchEvent Method
+
+```text
+t.dispatchEvent(target, eventName[, options]) → this | Promise<any>
+```
+
+> Important! The `t.dispatchEvent` method is not supported in Internet Explorer 11 and earlier.
+
+Fires a DOM event on the `target` element.
+
+Parameter              | Type                                              | Description
+---------------------- | ------------------------------------------------- | ----------------
+`target`               | Function &#124; String &#124; Selector &#124; Snapshot &#124; Promise | Identifies the event target. See [Select Target Elements](#select-target-elements).
+`eventName`            | String                                            | Event name. See [MDN Event Reference](https://developer.mozilla.org/en-US/docs/Web/Events)
```

Comment:
It's difficult to provide full list of events, since we can dispatch event with custom name.
Probably, we can make a list of EventsConstructors which are detected automatically based on event name

Reply:
The main idea - I should understand how to specify the `eventName` parameter.

> we can make a list of EventsConstructors which are detected automatically based on event name
From the point of view of a customer, it doesn't matter to me which event constructor is used

What about such describing style?
```js
A list of supported event names
You can specify any event name supported by browser. 
The most frequently used event names are
* mouse events (`mouseup`, `mousedown`, `click`, `dbclick`, `mousemove`, `mouseover`  `mouseleave`)
* keyboaed events (....)
* focus events
* input events
*....

Also if you use the CustomEvent constructor, you can specify an appropriate event name.

Reply:
> You can specify any event name supported by browser. 
You can specify any event name even if it does not supported by browser. Event name is just a name. You can dispatch event with name `foo`: `t.dispatchEvent('foo')`. It's valid way. However, in this case TestCafe will not recognize which event constructor should be used, so the CustomEvent constructor will be used.

Reply:
I see your point. I'll rewrite the piece with this in mind